### PR TITLE
DOC-3704: Rename xss linter.

### PR DESCRIPTION
**Pull request by robrap**
## [DOC-3704](https://openedx.atlassian.net/browse/DOC-3704)

To add clarity, the old "safe_template_linter.py" has been renamed to "xss_linter.py", and various updates were made to the documentation to use "XSS" or "Preventing XSS" in place of "Safe Templates", since this applies to more than just templates, and it applies only to XSS safety.

### Date Needed (optional)

This should be generally synced with the rename.

### Reviewers

Possible roles follow. The PR submitter checks the boxes after each reviewer finishes and gives :+1:. 

- [x] Subject matter expert: FYI: @robrap
- [x] Doc team review (sanity check, copy edit, or dev edit?): @edx/doc

### Testing

- [x] Ran ./run_tests.sh without warnings or errors

### HTML Version (optional)

- [ ] Build an RTD draft for your branch and add a link here

### Sandbox (optional)

- [ ] Point to or build a sandbox for the software change and add a link here

### Post-review

- [ ] Add a comment with the description of this change or link this PR to the next release notes task.
- [ ] Squash commits

### CODE REVIEW

**Comment by robrap** [comment]
@grantgoodmanedX: I'm not set up to run "./run_tests.sh" right now.  Should I get set up, or can you run that for me?  Thanks either way.

File: en_us/release_notes/source/2016/documentation/doc_0321_2016.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ Documentation for Open edX Users
 
 * Some sections of the *Open EdX Developer's Guide* have been reorganized. The
   "Writing Good Code" section now includes a new subsection,
-  :ref:`opendevelopers:Safe Templates`, as well as existing subsections
+  :ref:`opendevelopers:Preventing XSS`, as well as existing subsections
```
Comment (robrap) [inline en_us/release_notes/source/2016/documentation/doc_0321_2016.rst:8]: @grantgoodmanedX: I think this is failing with the following warning:
```
source/2016/documentation/doc_0321_2016.rst:6: WARNING: undefined label: opendevelopers:preventing xss (if the link has no caption the label must precede a section header)
source/2016/documentation/doc_0321_2016.rst:6: WARNING: undefined label: opendevelopers:preventing xss (if the link has no caption the label must precede a section header)
```
Not sure if I should even touch these old release notes or not, but it seems like they should point to the right place.  How do I fix?  Thanks.

Comment (grantgoodmanedX) [reply en_us/release_notes/source/2016/documentation/doc_0321_2016.rst:8]: I ran ./run_tests.sh. That warning should disappear once the changes are published. We can go ahead and merge this. 👍

Comment (robrap) [reply en_us/release_notes/source/2016/documentation/doc_0321_2016.rst:8]: Thanks @grantgoodmanedX.  I'm just waiting for the other changes to land first, and then I will let you know and you can merge/publish.

**Comment by robrap** [comment]
@grantgoodmanedX The dependent PRs have merged.  Can you merge this and update the docs when you get a chance?  Let me know too so I can update the wiki as well.  Thanks!

**Comment by robrap** [comment]
@grantgoodmanedX: FYI: I'll be out of the office until Wednesday.  The dependent changes merged, so don't wait on me to update the docs.  

Also, the following wiki page has one links that need to be updated once the new docs have been published:
https://openedx.atlassian.net/wiki/display/FEDX/Preventing+XSS+Best+Practices

Thank you!

**Pull request merged**